src/spi.rs

```diff
@@ -267,27 +280,29 @@ where
         self.wait_for_byte(ControlByte::Start as u8)
     }
 
-    fn read_and_check_byte(&mut self, check_byte: u8) -> Result<bool, self::Error> {
+    fn read_and_check_byte(&mut self, check_byte: &u8) -> Result<bool, self::Error> {
         match self.get_byte() {
             Ok(byte_out) => {
-                return Ok(byte_out == check_byte);
+                // Question: does comparing two &u8s work the way we would think?
+                return Ok(&byte_out == check_byte);
             }
             Err(e) => {
                 return Err(e);
             }
         }
     }
 
-    fn send_param<P: NinaParam>(&mut self, mut param: P) -> Result<(), self::Error> {
-        self.send_param_length(&mut param)?;
+    fn send_param<P: NinaParam>(&mut self, param: &P) -> Result<(), self::Error> {
+        self.send_param_length(param)?;
 
-        for byte in param.data().into_iter() {
-            self.bus.transfer(&mut [*byte]).ok().unwrap();
+        for byte in param.data().iter() {
+            // TODO: Try using `write_iter`: https://docs.rs/embedded-hal/latest/embedded_hal/blocking/spi/write_iter/index.html
```

Comment:
Do we still want to try `write_iter()` or happy with continuing to use `transfer()`?

Reply:
I think it makes sense to give `transfer()` a try since we have it working and we're growing in our understanding of how to continue to use the read write buffer API.
I removed the TODO in ff489b2a95ca35aef1ce4c24d396373e2d7838b4